zkevm-circuits/src/util/cell_manager.rs

```diff
@@ -1,8 +1,269 @@
-trait CellManagerStrategy {
-    fn query_top_aligned_cell(&mut self, cell_type: CellType) -> Cell<F>;
-    fn query_cell(&mut self, cell_type: CellType) -> Cell<F>;
-    fn query_cells(&mut self, cell_type: CellType, count: usize) -> Vec<Cell<F>>;
-    fn columns(&self) -> &[CellColumn<F>];
-    fn get_width(&self) -> usize;
+use std::collections::HashMap;
+
+use eth_types::Field;
+use gadgets::util::Expr;
+use halo2_proofs::{
+    circuit::{AssignedCell, Value},
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression, VirtualCells},
+    poly::Rotation,
+};
+
+use crate::{
+    evm_circuit::{table::Table, util::CachedRegion},
+    util::query_expression,
+};
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+/// CellType
```

Comment:
I know that a lot of these types and methods have been just moved from `zkevm-circuits/src/evm_circuit/util.rs` and were `pub(crate)` and didn't have much documentation.

Now many types and functions here are `pub` and you added some documentation lines which is just the type/function name.  I think it would be a very good opportunity to have a better documentation of the CellManager!  One or two lines for each type/function would be great.